### ISSUE TITLE
Deleting the entire index doesn't work

### ISSUE THREAD

**Issue by miskolc** [radar/elastic#7]
Normally I would expect that:

`ProductDocument.delete("/product") # where product is the @es_index`
 
To make a 
`
DELETE /product`

And ES to delete all the data in that index and respond with 

```
{
  "acknowledged": true
}
```

What actually happens is that no record gets deleted and I receive:
```

{:ok, 201,
 %{"_id" => "1", "_index" => "product",
   "_shards" => %{"failed" => 0, "successful" => 1, "total" => 2},
   "_type" => "product", "_version" => 1, "created" => true,
   "result" => "created"}}
```

I don't think this is consistent with the ES API and I'm not exactly sure why this happens.
I run 
`DELETE /product
`
in Kibana and
`curl -XDELETE "http://localhost:9200/product"
`
with cURL and both work as expected( delete the index).

Only the `delete` method from the ES package doesn't.

**Comment by radar** [comment]
I think you meant to use `ES.Index.delete("product")` here.

**Comment by miskolc** [comment]
No, I actually meant the `delete` method from the `Elastic.Document.API`( the one you import in your ProductDocument module). Now I see that there is indeed an `Elastic.Index.delete/1` method that does what I need, but was not present in the documentation.

**Comment by radar** [comment]
You are right. There was some missing documentation here. `Elastic.Index` was incorrectly configured with `@moduledoc false`. I've just released 2.3.5 which fixes this. You can see the documentation here: https://hexdocs.pm/elastic/Elastic.Index.html
